#! /usr/bin/env python

PACKAGE='controllers'

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

controller_enum = gen.enum([	gen.const("position", 		int_t, 	0, 	"Position control"),
				gen.const("velocity", 		int_t, 	1, 	"Velocity control"),
				gen.const("attitude", 		int_t, 	2, 	"Attitude control")
#				gen.const("att_velz_hybrid", 	int_t, 	3, 	"Attitude + velocity control")
				], "Controller type")

#       	Name			Type      	Level	Description				Default 	Min	Max 	Values
gen.add(	"controller", 		int_t,		0,	"Controller type",			1,		0,	3,	edit_method=controller_enum)
gen.add(	"thrust_multiplier", 	double_t,    	1,    	"Hover thrust is 1/multiplier", 	4.0,		1,	5)

exit(gen.generate(PACKAGE, "controllers", "setSafePX4"))
